### DISCUSSION TITLE
Lost main branch in Modern Farm Repository

### DISCUSSION THREAD

**Discussion by jrossgage** [nss-day-cohort-47/C47-Discussions#42] [Help Me] [answered]
Joel Gage

**Issue**
 When working on the modern farm, I pushed up my active branch to gitHub. When going to merge that branch with the main, I was not seeing the option to create a pull request. After some digging I could not find the main branch. It seems that my current active branch has replaced the main.

**Attempts**
I tried to just change the name of my active branch to main, however, it is saying that a main branch already exists. I can then checkout to that 'main' branch but when I open it up into VS Code, most of the files are missing and the files present say (deleted) next to them. 

**Googled**
https://gist.github.com/jbgo/1944238
https://stackoverflow.com/questions/50514697/git-lost-master-branch
https://stackoverflow.com/questions/3623755/why-does-my-git-branch-have-no-master

**GitHub links**
This issue is with pushing up to gitHub so I cannot provide direct github links at this time.

**Comment by BryanNilsen** [comment]
Your GitHub Modern Farm repository looks like it was updated 4 hours ago (at the time of me writing this)
It seems as if you made your initial commit today directly to main.

The other branch I see is jg-firstBranch and it's up to date with your main. So either you haven't written any code on this branch, or you pushed it without adding and committing any new code since you checked out from main.

I'm not sure what you have locally. Would need to see what branches you have and what's in them to really dig deeper into your issue.

**Comment by jrossgage** [comment, marked as answer]
:SOLVED: 
  I am not entirely certain how the mix up of branches occurred originally. I am still checking into that. However, there is an option on gitHub itself to alter the default branch of the repository. I created a new branch named 'main', then changed that branch to the default branch on gitHub. Thanks Monica for the help!
